web/packages/teleport/src/Roles/RoleEditor/RoleEditorAdapter.tsx

```diff
@@ -90,26 +98,29 @@ export function RoleEditorAdapter({
         {convertAttempt.status === 'error' && (
           <Danger>{convertAttempt.statusText}</Danger>
         )}
+        {roleDiffProps?.errorMessage && (
+          <Danger>{roleDiffProps.errorMessage}</Danger>
+        )}
         {convertAttempt.status === 'success' && (
           <RoleEditor
             originalRole={convertAttempt.data}
             onCancel={onCancel}
             onSave={onSave}
+            onRoleUpdate={onRoleUpdate}
           />
         )}
       </Flex>
-      <Flex flex="1" alignItems="center" justifyContent="center" m={3}>
-        {/* TODO (avatus) this component will not be rendered until the Access Diff feature is implemented */}
-        {roleDiffProps ? (
-          <roleDiffProps.RoleDiffComponent />
-        ) : (
+      {roleDiffProps ? (
+        roleDiffProps.roleDiffComponent
```

Comment:
Since we changed the semantics of the field, this is no longer a `roleDiffComponent`, but a `roleDiffElement` (see the [glossary](https://legacy.reactjs.org/docs/glossary.html#elements)).

Reply:
how does it fit now if its techincally a ReactNode?